File: CodeCheck/Published Standards/SEI Cert Standards/Java/SEC01-J/noncompliant_3.java
import java.io.*;
import java.nio.file.*;
import java.security.*;

public class Noncompliant3 {
  private void privilegedMethod(final String filename)
                                throws FileNotFoundException {
    try {
      Path path = (Path) AccessController.doPrivileged(
        new PrivilegedExceptionAction() {
          public Path run() throws FileNotFoundException {
            return Files.createTempDirectory(filename); // UndCC_Violation
          }
        }
      );
      // Do something with the file and then close it
    } catch (PrivilegedActionException e) {
      // Forward to handler
    }
  }
}

File: CodeCheck/Published Standards/SEI Cert Standards/C++/MEM51-CPP/Noncompliant3.cpp
struct P
{
};

class C
{
    P *p;

public:
    C(P *p) : p(p) {}
    ~C() { delete p; }  // UndCC_Violation

    void f() {}
};

void g(C c)
{
    c.f();
}

void h()
{
    P *p = new P;
    C c(p);
    g(c);
}
